mempool/aggregator.go

```diff
@@ -0,0 +1,78 @@
+package mempool
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/host"
+)
+
+// TODO: add String()
+type namespace int
+
+const (
+	Tendermint namespace = 1
+)
+
+type aggregator struct {
```

Comment:
So the design here is to make Mempool independent from namespaces/chain ids and have an aggregator that controls them.

I am wondering if we should make mempool on its own aware of multiple chains and make it part of API. E.g. you put a transaction into it and it reads the chain id from it and routes it to the correct queue/network. (Btw transactions should have a field on it to which chain they are related)

Reply:
Also, my intuition is to think about chain.Ids as like protocol.Ids which is a first-class citizen in the libp2p. Similarly, we want to make chain.Id/concod.ID to be the same.

Reply:
so your suggestion is to make a single mempool for different chainIds. Sounds good to me

Reply:
It will also help us to avoid namespaces